src/libstd/fs/mod.rs

```diff
@@ -752,6 +780,7 @@ pub fn set_file_times<P: AsPath + ?Sized>(path: &P, accessed: u64,
 /// This function will return an error if the provided `path` doesn't exist, if
 /// the process lacks permissions to change the attributes of the file, or if
 /// some other I/O error is encountered.
+#[stable(feature = "fs", since = "1.0.0")]
 pub fn set_permissions<P: AsPath + ?Sized>(path: &P, perm: Permissions)
```

Comment:
Hm, I wonder whether we should consider a builder-style API here instead?

It's a little unclear how extensive permissions will get, and whether in all cases you'll want to clobber the entire current permissions with an instance of `Permissions`. A builder could make it easier to say exactly what you want to override. Thoughts?